File: twopointers/28/sol.py
from my import Solution

if __name__ == "__main__":
    solution = Solution()
    
    # Example 1
    haystack1 = "sadbutsad"
    needle1 = "sad"
    result1 = solution.strStr(haystack1, needle1)
    print(f"Example 1: {result1} (Expected: 0)")
    
    # Example 2
    haystack2 = "leetcode"
    needle2 = "leeto"
    result2 = solution.strStr(haystack2, needle2)
    print(f"Example 2: {result2} (Expected: -1)")
    
    # Additional test - repeated pattern
    haystack3 = "mississippi"
    needle3 = "issip"
    result3 = solution.strStr(haystack3, needle3)
    print(f"Example 3: {result3} (Expected: 4)")
    
    # Edge case - single character
    haystack4 = "hello"
    needle4 = "l" 
    result4 = solution.strStr(haystack4, needle4)
    print(f"Example 4: {result4} (Expected: 2)")

File: twopointers/28/sol.c
#include <stdio.h>
#include <string.h>

int strStr(char * haystack, char * needle);

int main() {
    // Example 1
    char* haystack1 = "sadbutsad";
    char* needle1 = "sad";
    int result1 = strStr(haystack1, needle1);
    printf("Example 1: %d (Expected: 0)\n", result1);
    
    // Example 2
    char* haystack2 = "leetcode";
    char* needle2 = "leeto";
    int result2 = strStr(haystack2, needle2);
    printf("Example 2: %d (Expected: -1)\n", result2);
    
    // Additional test - repeated pattern
    char* haystack3 = "mississippi";
    char* needle3 = "issip";
    int result3 = strStr(haystack3, needle3);
    printf("Example 3: %d (Expected: 4)\n", result3);
    
    // Edge case - single character
    char* haystack4 = "hello";
    char* needle4 = "l";
    int result4 = strStr(haystack4, needle4);
    printf("Example 4: %d (Expected: 2)\n", result4);
    
    return 0;
}